test/components/elements/AutoCompleteMenuTest.js

```diff
@@ -0,0 +1,227 @@
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should call the select method once', function () {
+          expect(selectSpy).to.have.been.called.once
+        })
+      })
+
+    })
+
+    context('when the input is invalid', function() {
+      describe('when the input is an empty string', function () {
+        const dataSource = [{ title: 'Kermit' }, { title: 'Fozzie' }]
+        const input = ''
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('it should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should not call the select method', function () {
+          expect(selectSpy).to.not.have.been.called
+        })
+      })
+
+      describe('when the input entered matches more than one item in the data source', function() {
+        const dataSource = [{ name: 'Kermit' }, { name: 'Fozzie' }, {name: 'Fonzie'}]
+        const input = 'fo'
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should not call the select method', function () {
+          expect(selectSpy).to.not.have.been.called
+        })
+      })
+
+      describe('when the input entered is not in the data source', function () {
+        const dataSource = [{ name: 'Kermit' }, { name: 'Fozzie' }]
+        const input = 'l'
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should not call the select method', function () {
+          expect(selectSpy).to.not.have.been.called
+        })
+      })
+
+      describe('when the data source is empty', function () {
+        const dataSource = []
+        const input = 'l'
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('it should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should not call the select method', function () {
+          expect(selectSpy).to.not.have.been.called
+        })
+      })
+
+      describe('when the data source has invalid data', function () {
```

Comment:
How is this data source invalid?

Reply:
The method is looking for "id" and the data has "title" instead.  I'll clarify in the describe.